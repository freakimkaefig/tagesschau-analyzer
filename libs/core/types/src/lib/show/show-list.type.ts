import { SortDirection } from '../core.type';

export interface ShowListItem {
  date: Date;
  time: string;

  text: string;
}

export interface ShowListSortBy {
  key: string;
  direction: SortDirection;
}

export interface ShowListResponse {
  count: number;
  shows: ShowListItem[];
}
