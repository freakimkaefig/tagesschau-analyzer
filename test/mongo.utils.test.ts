import mockingoose from 'mockingoose';
import { ShowModel, IShowDocument } from '../src/models/show.model';

describe('test mongoose ShowModel', () => {
  it('should return the latest doc', async () => {
    const _doc = {
      _id: '507f191e810c19729de860ea',
      date: new Date('2020-06-20T20:00:00').toISOString(),
      time: '20-00',
      showId: 1234,
      text: 'Lorem ispum',
    };

    const finderMock = (query: any) => {
      expect(query.getQuery()).toMatchSnapshot('find query');

      if (query.getQuery().time === '20-00') {
        return _doc;
      } else {
        return {};
      }
    };

    mockingoose(ShowModel).toReturn(finderMock, 'find');

    return ShowModel.find({ time: '20-00' }).then((doc: IShowDocument[]) => {
      expect(JSON.parse(JSON.stringify(doc))).toMatchObject(_doc);
    });
  });

  it('should create returns mock', async () => {
    mockingoose(ShowModel).toReturn(
      { _id: '507f191e810c19729de860ea' },
      'save'
    );

    const result = await ShowModel.create({
      date: new Date('2020-06-20T20:00:00').toISOString(),
      time: '20-00',
      showId: 1234,
      text: 'Lorem ispum',
    });

    expect(JSON.parse(JSON.stringify(result))).toMatchObject({
      _id: '507f191e810c19729de860ea',
    });
  });
});
