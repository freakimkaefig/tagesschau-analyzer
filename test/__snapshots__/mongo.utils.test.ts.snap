// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test mongoose ShowModel should return the latest doc: find query 1`] = `
Object {
  "time": "20-00",
}
`;
